### PR TITLE
Configure webpack for dev environment

### PR DESCRIPTION
**Pull request by gkarat**
Fixes https://issues.redhat.com/browse/CCXDEV-5298

### CODE REVIEW

**Comment by gkarat** [comment]
Well, I tried to run `npm run start:beta:proxy` with the last commit from this PR. It seems custom proxy works OK with the Smart Proxy API, QA environment. I have tried couple of endpoints locally (e.g. GET /groups).

**Comment by gkarat** [comment]
@AllenBW, could you please check it ooout? Something may be missing; thus, I want to have your expert eye on it 😸

**Comment by gkarat** [comment]
I've also run into one issue I haven't found a way how to solve yet: when opening `/beta/openshift/insights/advisor` it redirects you to `/beta/openshift/insights/sample` instead of desired `/beta/openshift/insights/advisor/sample`.

File: config/dev.webpack.config.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const { config: webpackConfig, plugins } = config({
   customProxy: process.env.API_ENDPOINT
     ? [
         {
-          context: ['/api'],
+          context: (path) => path.includes('/api/'),
```
[the remark this answers is not in the document]
Comment (AllenBW) [reply config/dev.webpack.config.js:13]: one of those situations where yah gotta blend documentation from ah few sources, check it out https://webpack.js.org/configuration/dev-server/#devserverproxy

**Pull request merged**